### ISSUE TITLE
Research ToggleButtons

### ISSUE THREAD

**Issue by lukewarlow** [openui/open-ui#957]
Toggle buttons are pretty common UI elements that currently require aria (aria-pressed). I feel these would be relatively simple to specify a native ability for.

First thing to do would be pull together research on existing design systems to work out existing implementations. (https://react-spectrum.adobe.com/react-aria/ToggleButton.html is just one example)

**Comment by lukewarlow** [comment]
Name tbd obviously (would probably go with `<button type="toggle">` or `<button type="button" toggle>`).

But then maybe we want a pressed attribute to take the true false mixed values.

Which begs the question do we want tri-state?

All of this would hopefully be answered by research.

**Comment by lukewarlow** [comment]
I propose this would be a valid element for invoketarget (like a button with type="button" and type="reset")

**Comment by lukewarlow** [comment]
These along with `focusgroup` would allow for a proper toolbar implementation such as the one on the GitHub comments panel (or really any Toolbar).

Also useful for UI such as in video calls where you have UI for toggling microphone and camera state (though it's key that we educate to ensure people don't change the label as part of changing the pressed state).

**Comment by scottaohara** [comment]
@lukewarlow i had been tooling around with a draft proposal for this a couple weeks ago. I would argue that instead of introducing a new attribute, we reuse the `checked` attribute.  

unfortunately 'toggle' (in this context indicating a pressed/not pressed or mixed state) could get confusing with 'toggling' a popover/dialog - since those are also toggling but not self toggling.  sigh.  we need more words rather than calling everyting a tooltip, toggle or a card.

**Comment by lukewarlow** [comment]
Yeah thinking on it I think pressed (to match aria?) or checked or something other than toggle is for the best.

**Comment by bkardell** [comment]
I agree either of those is better than toggle or toggled or whatever.

**Comment by lukewarlow** [comment]
I've raised some issues to try and gather some use cases.


https://github.com/openui/open-ui/issues/1039 - toggle buttons
https://github.com/openui/open-ui/issues/1040 - cycle buttons (the sort of thing Keith mentions above)

**Comment by lukewarlow** [comment]
Closing this given we've got press button explainer, if anyone else wants to pursue the cycle buttons feel free too.
